ext/fs/std_fs.rs

```diff
@@ -821,6 +821,43 @@ fn stat_extra(
     Ok(info.dwVolumeSerialNumber as u64)
   }
 
+  unsafe fn get_change_time(
+    handle: winapi::shared::ntdef::HANDLE,
+  ) -> std::io::Result<u64> {
+    use winapi::shared::minwindef::FALSE;
+    use winapi::um::fileapi::FILE_BASIC_INFO;
+    use winapi::um::minwinbase::FileBasicInfo;
+    use winapi::um::winbase::GetFileInformationByHandleEx;
+
+    let file_info = {
+      let mut file_info = std::mem::MaybeUninit::<FILE_BASIC_INFO>::zeroed();
+      if GetFileInformationByHandleEx(
```

Comment:
Are you sure about this? Libuv (which is what Node uses under the hood) gets this information from the `NtQueryInformationFile` syscall: https://github.com/libuv/libuv/blob/4e310d0f90af29e699e2dedad5fa0dcee181a7cc/src/win/fs.c#L1751-L1793

I think it's quite unfortunate that we would have to do another syscall just to obtain the ctime value.

Reply:
So I did some digging, and here are my findings, it would be great if someone could check it on their machine, too.

In my case, at least the `NtQueryInformationFile` doesn't work completely!
I checked the status code for this call. It always returns some big (which might not matter) negative number meaning that it failed to get the data, but on the other hand `std::io::Error::last_os_error()` returns an object with a status code of 0 and information that it was successful.

So I found that we close a file handle in code before actually using it in the `query_file_information ` method.
![image](https://github.com/user-attachments/assets/e1029026-2954-418b-85c4-6d5857ef3436)
but even moving it below the call doesn't change anything at all and `NtQueryInformationFile` still fails the same way as before.

Now I tried to use the `NtQueryInformationFile` method in just plain rust program and even tried to get only basic info instead of the full one, but it also failed. I tried that on two different machines with Windows 11 64bit and ARM on board, and it doesn't work in both.

If you could check on your end whether this is only my machine or if this doesn't seem to work at all, that would be great.

Reply:
`NtQueryInformationFile` doesn't set last_os_error() upon failure. Instead, it returns an NTSTATUS code and you have to convert it to a win32 error (using [RtlNtStatusToDosError](https://learn.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-rtlntstatustodoserror)) manually to know what the error was.

Reply:
Good to know, thanks, I assumed this cause that's how handling is done in deno code right now. I will have a look at this then, what about the CloseHandle placement, shouldn't it be after each call that uses the handle?

Reply:
It looks like the call to `RtlNtStatusToDosError` is entirely missing from the current code base, which means that when `NtQueryInformationFile` fails we are not currently reporting an appropriate error code.

Reply:
Thank you so much for your help! I'm not an expert on a Windows API topic (it is my first time touching it), but it was a lot of fun to learn about it! 😄. I managed to fix the `NtQueryInformationFile` call and get all the data from a single call, and I have added some proper error handling. Let me know if there is something else that can be improved.